### PR TITLE
Bump required ES version

### PR DESCRIPTION
**Pull request by im-denisenko**
Also added some of `@deprecated` docblocks

### CODE REVIEW

**Comment by ruflin** [comment]
@im-denisenko Let me know when this is ready to be merged, not that I merge it too early. 

BTW: Nice list https://github.com/ruflin/Elastica/issues/812

File: changes.txt [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+- Added Elastica\Query\SimpleQueryString::setMinimumShouldMatch
+- Added Elastica\Query\FunctionScore::setMinScore
+- Added Elastica\Query\MoreLikeThis::setMinimumShouldMatch
+
+2015-04-08
+- Added new methods to Elastica\Aggregation\DateHistogram: setOffset, setTimezone
+- Following methods in Elastica\Aggregation\DateHistogram marked as deprecated: setPreOffset, setPostOffset, setPreZone, setPostZone, setPreZoneAdjustLargeInterval
+
+2015-04-07
+- [BC break] Elastica\Query\QueryString::setLowercaseExpandedTerms removed
```
Comment (ruflin) [inline changes.txt]: Do you know in which version of ES this was removed? I didn't check the changelog ...

Comment (im-denisenko) [inline changes.txt]: It was removed [here](https://github.com/elastic/elasticsearch/pull/10086) and mentioned in changelog of [1.5.0](https://www.elastic.co/downloads/past-releases/elasticsearch-1-5-0).

Comment (ruflin) [inline changes.txt]: thx

**Comment by ruflin** [comment]
@im-denisenko I assume this pull request is ready to be merged?

**Comment by im-denisenko** [comment]
@ruflin Sorry, I was quite busy recently. Yes, let's close it.

**Comment by ruflin** [comment]
Merged. Thx.

**Pull request merged**